update-dependencies/Program.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.VersionTools;
+using Microsoft.DotNet.VersionTools.Automation;
+using Microsoft.DotNet.VersionTools.Dependencies;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Dotnet.Docker.Nightly
+{
+    public static class Program
+    {
+        private static readonly string s_repoRoot = Directory.GetCurrentDirectory();
+        private static readonly Config s_config = Config.s_Instance;
+        private static bool s_updateOnly = false;
+
+        public static void Main(string[] args)
+        {
+            if (ParseArgs(args))
+            {
+                DependencyUpdateResults updateResults = UpdateFiles();
+
+                if (!s_updateOnly && updateResults.ChangesDetected())
+                {
+                    CreatePullRequest(updateResults).Wait();
+                }
+            }
+        }
+        
+        private static bool ParseArgs(string[] args)
+        {
+            foreach (string arg in args)
+            {
+                if (string.Equals(arg, "--Update", StringComparison.OrdinalIgnoreCase))
+                {
+                    s_updateOnly = true;
+                }
+                else if (arg.Contains('='))
+                {
+                    int delimiterIndex = arg.IndexOf('=');
+                    string name = arg.Substring(0, delimiterIndex);
+                    string value = arg.Substring(delimiterIndex + 1);
+
+                    Environment.SetEnvironmentVariable(name, value);
+                }
+                else
+                {
+                    Console.Error.WriteLine($"Unrecognized argument '{arg}'");
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static DependencyUpdateResults UpdateFiles()
+        {
+            // Update the version of the .NET Core SDK in the Dockerfiles to the latest version in the
+            // specified CLI channel. This version is determined by using DotNet.VersionTools to examine the
+            // latest version of one of the packages produced by the CLI build (Microsoft.DotNet.Cli.Utils),
+            // which has the same version as the SDK itself.
```

Comment:
My intent was not to reiterate what the code does.  I think the code is readable already.  My point was to simply explain why the code was grabbing the Microsoft.DotNet.Cli.Utils version versus the CLI "Product".  

"The BuildInfo does not contain the CLI product version therefore the version is retrieved from a particular CLI package (eg. Microsoft.DotNet.Cli.Utils).  Once the BuildInfo includes the product version, it should be utilized."